snapcraft/internal/build_providers/_lxd/_lxd.py

```diff
@@ -0,0 +1,277 @@
+# -*- Mode:Python; indent-tabs-mode:nil; tab-width:4 -*-
+#
+# Copyright (C) 2019 Canonical Ltd
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License version 3 as
+# published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+import logging
+import os
+import subprocess
+import sys
+from typing import Optional, Sequence
+
+import pylxd
+
+from .._base_provider import Provider
+from .._base_provider import errors
+from ._images import get_image_source
+from snapcraft.internal import repo
+from snapcraft.internal.errors import SnapcraftEnvironmentError
+
+
+logger = logging.getLogger(__name__)
+
+
+class LXD(Provider):
+    """A LXD provider for snapcraft to execute its lifecycle."""
+
+    _PROJECT_DEVICE_NAME = "snapcraft-project"
+
+    @classmethod
+    def ensure_provider(cls):
+        try:
+            if repo.snaps.SnapPackage.is_snap_installed("lxd"):
+                return
+        except repo.errors.SnapdConnectionError as snapd_error:
+            if sys.platform == "linux":
+                error_message = "snap support is required to continue: {}".format(
+                    snapd_error
+                )
+            else:
+                error_message = "LXD is not supported on this platform"
```

Comment:
How about:
>LXD is not supported on this platform. Use multipass to run snapcraft on non-Linux systems: https://github.com/CanonicalLtd/multipass

Reply:
Yeah, that looks good, although we could also tell people to just run without --use-lxd, what do you think?

Reply:
I am not sure about this one now... this might be the wrong way to solve this.

Reply:
I will fix in a different PR provided I get a bug for this :smile:
But the gist of it, is that we should solve this at a higher layer to keep a good separation of concerns.

Reply:
https://bugs.launchpad.net/snapcraft/+bug/1822545